src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipFileProcessor.java

```diff
@@ -39,7 +40,11 @@ public ZipFileProcessor(Envelope envelope) {
         this.envelope = envelope;
     }
 
-    public void process(ZipInputStream zis) throws IOException {
+    public void process(
```

Comment:
Process method should ideally handle the extraction and if verification enabled call the zip verifier util class. 
Code would be much easier to read that way imo.

Reply:
I could not disagree more :-) as you'd be creating a dependency between the extraction and all the extra activities which should just be pluggable.

Reply:
your verifier class is currently extracting the zip file. at least that logic should be in some separate class.

Reply:
The signature and zipped documents are in a zip. To verify the signature you need to extract that. If you put that logic in a separate class you have split the verifier and made it dependent on the zip documents processor. I really would not do that.

Reply:
Verifier is a util class so why it would be a dependent ?